src/TorchSharp/TorchVision/Image.cs

```diff
@@ -0,0 +1,177 @@
+using System.IO;
+using System.Threading.Tasks;
+
+using static TorchSharp.torch;
+
+namespace TorchSharp.torchvision
+{
+    public static partial class io
+    {
+        /// <summary>
+        /// <cref>Imager</cref> to be used when a <cref>torchvision.io</cref> image method's <c>imager</c> is unspecified.
+        /// </summary>
+        public static Imager DefaultImager { get; set; }
+
+        /// <summary>
+        /// Abstract class providing a generic way to decode and encode images as <cref>Tensor</cref>s.
+        /// Used by <cref>torchvision.io</cref> image methods.
+        /// </summary>
+        public abstract class Imager
+        {
+            /// <summary>
+            /// Reads the contents of an image file and return the image's file format.
+            /// </summary>
+            /// <param name="image">Image file contents.</param>
+            /// <returns>The detected image format.</returns>
+            public abstract ImageFormat DetectFormat(byte[] image);
```

Comment:
I think this method should disappear. In the SkiaImage example, it requires an additional copy of the data. It's better to have DecodeImage decide how it's going to figure out what format it is in the best way for each implementation. It can still be a private method on the ImageSharp implementation.

Reply:
That means making all the DecodeImage methods abstract.